### DISCUSSION TITLE
Making a subset object in Xenium analysis

### DISCUSSION THREAD

**Discussion by NA0004** [satijalab/seurat#7947] [Q&A]
Hi,

I would like to know how to subset Xenium object.

Following the tutorial "Analysis of Image-based Spatial Data in Seurat", I cropped xenium.obj using the tutorial codes.

cropped.coords <- [Crop](https://mojaveazure.github.io/seurat-object/reference/Crop.html)(xenium.obj[["fov"]], x = [c](https://rdrr.io/r/base/c.html)(1200, 2900), y = [c](https://rdrr.io/r/base/c.html)(3750, 4550), coords = "plot")
xenium.obj[["zoom"]] <- cropped.coords

Now I would like to make a new Xenium object of this area by using subset function so that I can analyze specific brain area and DEG between two areas using FindMarker function.

I have tried to use subset command mimicking visium tutorial found on the web, but I have been failing.

It would be very helpful if I could get help for subsetting Xenium object.

Thank you.

**Comment by gretajuu** [comment]
Hello,

Were you able to subset the data? I would be interested of this feature too.

**Comment by NA0004** [comment]
Hi,

yes, I could subset data!

It would be great if Seurat could provide tools to make susets!

Best,
NA

2024年3月11日(月) 9:44 gretajuu ***@***.***>:

**Comment by kelvinmo0513** [comment]
Hi @NA0004,

Could you please let me know how you subset the xenium data?

Thanks,
Kelvin

**Comment by ilovegreentea** [comment]
Hey

I used subset_opt function

https://github.com/alikhuseynov/add-on_R/blob/develop/R/subset_obj_seurat.R

**Comment by kelvinmo0513** [comment]
Thanks so much for sharing the function with me @ilovegreentea !!
This is my code and does this function work to subset the xenium object based on coordinates of cropped image?

crop <- Crop(xenium.obj[["fov"]], x = c(-96.42, 3370.79), y = c(1162.64, 3130.46))
xenium.obj[["cropped"]] <- crop

**Comment by ilovegreentea** [comment]
Hi Kelvin,

I'm not sure, I understand your question well.
If you added fov to a xenium object before subset,
It also works in cropped images.



2024년 5월 20일 (월) 오전 3:24, Kelvin Mo ***@***.***>님이 작성:

**Comment by kelvinmo0513** [comment]
Hi @ilovegreentea 

Sorry for the confusion. So what I did was to crop the image from the xenium data and stored it in xenium.obj[["cropped"]]. However, I was not able to access the metadata of the cropped subset. Can I know how would you use the _subset_opt_ function to subset the xenium.obj based on the cropped image?

Thanks,
Kelvin

**Comment by ilovegreentea** [comment]
Oh I got it.

I'm going to send my code.
I used barcode information in ***@***.******@***.***

xenium_obj<- Crop( xenium_obj  [["fov"]], x = c(300, 2500), y = c(000,
2300), coords = "plot")
xenium_obj[[" xenium_obj _1"]] <-  xenium_obj _1
***@***.***$sample_name <- ifelse(  xenium_obj @meta.data$barcode
%in%  xenium_obj ***@***.******@***.***, "annotation" ,
***@***.***$sample_name )

I hope I get it right this time.

Thanks,
KyuTae

2024년 5월 20일 (월) 오후 2:56, Kelvin Mo ***@***.***>님이 작성:

**Comment by kelvinmo0513** [comment]
Thank you so much @ilovegreentea ! This is helpful.

One side question: After you crop the image and place it in xenium.obj[["xenium_obj_1"]], how do you access the cell identities or barcode information? For me, it's just an image under the "fov" of the xenium object without any info.

**Comment by ilovegreentea** [comment]
Why is my code obscured??
***@***.******@***.***


2024년 5월 20일 (월) 오후 3:17, Kelvin Mo ***@***.***>님이 작성:

**Reply by ilovegreentea** [reply]
xenium_obj_1(at sign)boundaries$centroids(at sign)cells

**Reply by kelvinmo0513** [reply]
Omg thank you so much! This solves my issue. You're a life saver:)

**Reply by aksaikumarjayalatha** [reply]
Hi I am sorry but I am unable to follow these codes as they're obsecured. For the same setup of Kelvin, how can I subset my Xenium data, using the cropped coordinates information? 
From this code `xenium_obj<- Crop( xenium_obj  [["fov"]], x = c(300, 2500), y = c(000,
2300), coords = "plot")
xenium_obj[[" xenium_obj _1"]] <-  xenium_obj _1
***@***.***$sample_name <- ifelse(  xenium_obj @meta.data$barcode
%in%  xenium_obj ***@***.******@***.***, "annotation" ,
***@***.***$sample_name )`

What is this ***@***.***$sample_name?

Thank you!

**Reply by babadriskobaba-boop** [reply]
Did anyone wants to get my zangi number

**Comment by HansLambda** [comment]
You can use the standard subset function for Seurat objects. Once the cropped image is stored as "zoom", just pull out the cells.
```
cropped.cells <- xenium.obj@images$zoom$centroids@cells
cropped.obj <- subset(xenium.obj, cells = cropped.cells)
```

**Comment by msraredon** [comment]
My solution to this has been to build a new metadata column just for this operation, because subsetting on metadata works just fine. 

So, if we have a big object called 'sub' and we want to make a small object called 'demo', defined by a custom FOV/crop:

## define small crop
cropped.coords <- Crop(sub[["PNX3_M2_Sp"]], 
                       x = c(6900, 7075), 
                       y = c(4450, 4600), 
                       coords = "plot")

## set a new field of view (fov)
sub[["demo"]] <- cropped.coords

# isolate demo into solitary object
## initialize new metadata column
sub$demo.cells <- NA
## identify cells to pull
cropped.cells <- sub@images$demo$centroids@cells
## annotate the empty metadat acolumn to identify these desired cells as a group
sub$demo.cells[cropped.cells] <- 'demo'
## subset based on this group value within the new metadata column
demo <- subset(sub, subset = demo.cells == 'demo')
## inspect output
demo

**Comment by babadriskobaba-boop** [comment]
Hey so cute
